### PR TITLE
Rest client assault

### PR DESCRIPTION
**Pull request by fletchgqc**
**What**:

**Why**:

**How**:

**Checklist**:

- [ ] Documentation added
      
- [ ] Changelog updated
      
- [ ] Tests added
      
- [ ] Ready to be merged

### CODE REVIEW

File: chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/assaults/ResponseModifyingRequestInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      return 500;
+    }
+
+    @Override
+    public String getStatusText() throws IOException {
+      return HttpStatus.INTERNAL_SERVER_ERROR.getReasonPhrase();
+    }
+
+    @Override
+    public void close() {
```
Comment (WtfJoke) [inline chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/assaults/ResponseModifyingRequestInterceptor.java:46]: I think you should close the stream from getBody

File: chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/assaults/WebClientAssaultTest.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.ExchangeFilterFunction;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import reactor.core.publisher.Mono;
+
+/** @author Benjamin Wilms */
```
Comment (WtfJoke) [inline chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/assaults/WebClientAssaultTest.java:35]: I think you should remove that line :D

File: chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/assaults/RestTemplateAssaultTest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  @Test
+  void assaultRestTemplateCallWithLatency() {
+    restTemplate.setInterceptors(
+        Collections.singletonList(new LatencyInducingRequestInterceptor()));
+    Instant start = Instant.now();
+    restTemplate.getForEntity("https://jsonplaceholder.typicode.com/todos/1", String.class);
+    Instant end = Instant.now();
+
+    Duration d = Duration.between(start, end);
```
Comment (WtfJoke) [inline chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/assaults/RestTemplateAssaultTest.java:56]: I think you either should give that variable a better name and/or use `long requestTimeInMillis = Duration.between(start,end).toMillis());`

File: chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/assaults/WebClientAssaultTest.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        .get()
+        .uri("https://jsonplaceholder.typicode.com/todos/1")
+        .accept(MediaType.APPLICATION_JSON)
+        .retrieve()
+        .bodyToMono(String.class)
+        .block();
+
+    Instant end = Instant.now();
+
+    Duration d = Duration.between(start, end);
```
Comment (WtfJoke) [inline chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/assaults/WebClientAssaultTest.java:103]: Same as before :)

File: chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/assaults/RestTemplateAssaultTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import java.util.Collections;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.data.Offset;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+/** @author Benjamin Wilms */
```
Comment (WtfJoke) [inline chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/assaults/RestTemplateAssaultTest.java:30]: ➖

File: chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/assaults/RestTemplateAssaultTest.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+@ExtendWith(MockitoExtension.class)
+class RestTemplateAssaultTest {
+
+  private RestTemplate restTemplate = new RestTemplate();
+
+  @Test
+  void assaultRestTemplateCallWithFailure() {
+    restTemplate.setInterceptors(
+        Collections.singletonList(new ResponseModifyingRequestInterceptor()));
+    Assertions.assertThatThrownBy(
```
Comment (WtfJoke) [inline chaos-monkey-spring-boot/src/test/java/de/codecentric/spring/boot/chaos/monkey/assaults/RestTemplateAssaultTest.java:40]: I would suggest using static imports for asssertJ methods

**Review by WtfJoke — CHANGES_REQUESTED** [review]
Some smaller stuff, but in general it looks promising 👍

**Comment by fletchgqc** [comment]
Thanks. This is just a POC, not mergeable code. That’s why it’s a  PR to my own repo.

**Pull request closed without merging**